ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/target/VirtualArtifactRepository.java

```diff
@@ -123,8 +127,17 @@ private void transferArtifact(Object artifactModel, OutputStream destination) th
 			if (location == null) {
 				throw new FileNotFoundException(bundleInfo.getSymbolicName());
 			}
-			try (InputStream is = location.toURL().openStream()) {
-				is.transferTo(destination);
+
+			File bundleLocation = new File(location);
+			if (bundleLocation.isFile()) {
+				try (InputStream is = new FileInputStream(bundleLocation)) {
+					is.transferTo(destination);
+				}
+			} else {
+				ZipOutputStream zip = new ZipOutputStream(destination);
+				FileUtils.zip(zip, bundleLocation, Set.of(), FileUtils.createRootPathComputer(bundleLocation));
+				zip.finish();
+				zip.flush();
```

Comment:
I think this flush() is not necessary.
```suggestion
```

Reply:
When the above change is made to make things consistent, this can be removed.

Perhaps @ptziegler would like to have a closer look?

Reply:
> I think this flush() is not necessary.

Why do you think this and why would it harm to explicitly flush the content?